app/services/checksum_validator.rb

```diff
@@ -1,12 +1,112 @@
 # code for validating Moab checksums
 class ChecksumValidator
-  def initialize(druid, endpoint, algorithm)
-    @druid = druid
-    @endpoint = endpoint
-    @algorithm = algorithm
+
+  attr_reader :handler_results, :druid, :endpoint
+
+  MANIFESTS = 'manifests'.freeze
+  MANIFESTS_XML = 'manifestInventory.xml'.freeze
+
+  def initialize(druid, storage_dir)
+    @druid = "druid:#{druid}"
+    @endpoint = Endpoint.find_by(storage_location: storage_dir)
+    @handler_results = AuditResults.new(druid, nil, endpoint)
   end
 
   def validate_checksum
     # TODO: implement this;  we begin with a placeholder
   end
+
+  # TODO: remove this once refactor parse_verification_subentity
+  def validate_manifest_inventories
+    storage_object.version_list.each do |storage_object_version|
+      validate_manifest_inventory(storage_object_version)
+    end
+    handler_results.report_results
+  end
+
+  private
+
+  def validate_manifest_inventory(storage_object_version)
+    manifest_file_path = "#{storage_object_version.version_pathname}/#{MANIFESTS}/#{MANIFESTS_XML}"
+    begin
+      verification_result = storage_object_version.verify_manifest_inventory
+      parse_verification_subentities(verification_result)
+    rescue Nokogiri::XML::SyntaxError
+      handler_results.add_result(AuditResults::INVALID_MANIFEST, manifest_file_path: manifest_file_path)
+    rescue Errno::ENOENT
+      handler_results.add_result(AuditResults::MANIFEST_NOT_IN_MOAB, manifest_file_path: manifest_file_path)
+    end
+  end
+
+  def parse_verification_subentities(result)
```

Comment:
given that we have AuditResults object, can we name this variable something clearer?   manifest_inventory_verification_result?  manifest_inventory_verification_output?